templates/types/streaming/fastapi/app/api/routers/chat.py

```diff
@@ -1,32 +1,52 @@
-import os
 import logging
+import os
+from typing import List
 
-from aiostream import stream
-from fastapi import APIRouter, Depends, HTTPException, Request, status
+from fastapi import APIRouter, BackgroundTasks, Depends, HTTPException, Request, status
+from llama_index.core.chat_engine.types import BaseChatEngine, NodeWithScore
 from llama_index.core.llms import MessageRole
-from llama_index.core.chat_engine.types import BaseChatEngine
-from llama_index.core.vector_stores.types import MetadataFilters, MetadataFilter
-from app.engine import get_chat_engine
-from app.api.routers.vercel_response import VercelStreamResponse
+from llama_index.core.vector_stores.types import MetadataFilter, MetadataFilters
+
+from app.api.controllers.llama_cloud import LLamaCloudFileController
 from app.api.routers.events import EventCallbackHandler
 from app.api.routers.models import (
-    ChatData,
     ChatConfig,
-    SourceNodes,
-    Result,
+    ChatData,
     Message,
+    Result,
+    SourceNodes,
 )
+from app.api.routers.vercel_response import VercelStreamResponse
+from app.engine import get_chat_engine
 
 chat_router = r = APIRouter()
 
 logger = logging.getLogger("uvicorn")
 
 
+def process_response_nodes(
+    nodes: List[NodeWithScore],
+    background_tasks: BackgroundTasks,
+):
+    """
+    Start background tasks on the source nodes if needed.
+    """
+    files_to_download = SourceNodes.get_download_files(nodes)
+    for file in files_to_download:
+        background_tasks.add_task(
+            LLamaCloudFileController.download_llamacloud_pipeline_file,
+            file_name=file.get("file_name"),
+            pipeline_id=file.get("pipeline_id"),
+        )
```

Comment:
isn't it `file` as paramter?
```suggestion
        background_tasks.add_task(
            LLamaCloudFileController.download_llamacloud_pipeline_file,
            file
        )
```